rootfs/etc/nginx/lua/balancer.lua

```diff
@@ -1,86 +1,68 @@
 local ngx_balancer = require("ngx.balancer")
 local json = require("cjson")
 local configuration = require("configuration")
-local util = require("util")
-local lrucache = require("resty.lrucache")
+local round_robin = require("balancer.round_robin")
+local chash = require("balancer.chash")
+local sticky = require("balancer.sticky")
 local ewma = require("balancer.ewma")
-local resty_balancer = require("balancer.resty")
 
 -- measured in seconds
 -- for an Nginx worker to pick up the new list of upstream peers
 -- it will take <the delay until controller POSTed the backend object to the Nginx endpoint> + BACKENDS_SYNC_INTERVAL
 local BACKENDS_SYNC_INTERVAL = 1
 
 local DEFAULT_LB_ALG = "round_robin"
+local IMPLEMENTATIONS = {
+  round_robin = round_robin,
+  chash = chash,
+  sticky = sticky,
+  ewma = ewma,
+}
 
 local _M = {}
+local balancers = {}
 
--- TODO(elvinefendi) we can probably avoid storing all backends here. We already store them in their respective
--- load balancer implementations
-local backends, backends_err = lrucache.new(1024)
-if not backends then
-  return error("failed to create the cache for backends: " .. (backends_err or "unknown"))
-end
-
-local function get_current_backend()
-  local backend_name = ngx.var.proxy_upstream_name
-  local backend = backends:get(backend_name)
-
-  if not backend then
-    -- TODO(elvinefendi) maybe force backend sync here?
-    ngx.log(ngx.WARN, "no backend configuration found for " .. tostring(backend_name))
-  end
-
-  return backend
-end
+local function get_implementation(backend)
+  local name = backend["load-balance"] or DEFAULT_LB_ALG
 
-local function get_balancer(backend)
-  if not backend then
-    return nil
+  if backend["sessionAffinityConfig"] and backend["sessionAffinityConfig"]["name"] == "cookie" then
+    name = "sticky"
```

Comment:
If a backend `load-balance` algorithm is set, and is overriden here since `sessionAffinity` is required - should we emit a log message here indicating that the load-balance implementation is being ignored?

Reply:
This is a default behaviour in non dynamic mode as well. We can maybe document better how this annotation overrides `load-balance` annotation and config, but I don't see how logging here would add any value.

Reply:
> That being said we can do INFO logging if you think it is valuable. But I'd rather see this becoming necessity in practice first and then add the logging.

 💯